distributions/test/itests/src/test/java/org/codice/ditto/replication/admin/query/ITReplicationQuery.java

```diff
@@ -0,0 +1,190 @@
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.core.Is.is;
+
+import com.jayway.restassured.specification.RequestSpecification;
+import java.net.MalformedURLException;
+import org.junit.Test;
+
+public class ITReplicationQuery {
+
+  private static final String GRAPHQL_ENDPOINT = "https://localhost:8993/admin/hub/graphql";
+
+  private static final String URL = "https://localhost:9999";
+
+  // ----------------------------------- Site Tests -----------------------------------//
+
+  @Test
+  public void createReplicationSite() throws MalformedURLException {
+    String name = "create";
+    asAdmin()
+        .body(makeCreateSiteQuery(name, URL))
+        .when()
+        .post(GRAPHQL_ENDPOINT)
+        .then()
+        .statusCode(200)
+        .body("data.createReplicationSite.name", is(name))
+        .body("data.createReplicationSite.address.url", is(URL))
+        .body("data.createReplicationSite.address.host.hostname", is("localhost"))
+        .body("data.createReplicationSite.address.host.port", is(9999))
+        .body("data.createReplicationSite.id", is(notNullValue()));
+  }
+
+  @Test
+  public void createReplicationSiteWithHostnameAndPort() {
+    String name = "createHostnameAndPort";
+    asAdmin()
+        .body(
+            String.format(
+                "{\"query\":\"mutation{ createReplicationSite(name: \\\"%s\\\", address: { host: { hostname: \\\"localhost\\\" port: 9999 }}){ id name address{ host{ hostname port } url }}}\"}",
+                name))
+        .when()
+        .post(GRAPHQL_ENDPOINT)
+        .then()
+        .statusCode(200)
+        .body("data.createReplicationSite.name", is(name))
+        .body("data.createReplicationSite.address.url", is(URL))
+        .body("data.createReplicationSite.address.host.hostname", is("localhost"))
+        .body("data.createReplicationSite.address.host.port", is(9999))
+        .body("data.createReplicationSite.id", is(notNullValue()));
+  }
+
+  @Test
+  public void createReplicationSiteWithEmptyName() {
+    asAdmin()
+        .body(makeCreateSiteQuery("", URL))
+        .when()
+        .post(GRAPHQL_ENDPOINT)
+        .then()
+        .statusCode(200)
+        .body("errors.message", hasItem("EMPTY_FIELD"));
+  }
+
+  @Test
+  public void createReplicationSiteWithInvalidUrl() {
+    asAdmin()
+        .body(makeCreateSiteQuery("badUrl", "localhost:9999"))
+        .when()
+        .post(GRAPHQL_ENDPOINT)
+        .then()
+        .statusCode(200)
+        .body("errors.message", hasItem("INVALID_URL"));
+  }
+
+  @Test
+  public void createReplicationSiteWithIntegerName() {
+    asAdmin()
+        .body(
+            String.format(
+                "{\"query\":\"mutation{ createReplicationSite(name: 25, address: { url: \\\"%s\\\"}){ id name address{ host{ hostname port } url }}}\"}",
+                URL))
+        .when()
+        .post(GRAPHQL_ENDPOINT)
+        .then()
+        .statusCode(200)
```

Comment:
❗️ I would say returning 200 OK when a request is invalid is a problem. This is not something a client of an API would expect.

Reply:
When type validation fails, GraphQL returns a 200 but the "errors" field in the JSON response is populated.

Reply:
I would think that a 400 would make more sense, but I think that is a problem with the endpoint, not with the tests.

Reply:
it depends. I agree that 400 makes more sense to me as well. If that is what we say our API should return then we should validate that in our test and let the test fail until we fix it. If it turns out that 200 is valid in this case as we specify an error code inside the response message than the test case is ok. It all depends on what we document our API/protocol to be.